docs/tutorials/03_shaders/viz_sdf_cylinder.py

```diff
@@ -0,0 +1,296 @@
+# cylinders, and add them to the scene.
+
+centers = np.array([[-3.2, .9, .4], [-3.5, -.5, 1], [-2.1, 0, .4],
+                    [-.2, .9, .4], [-.5, -.5, 1], [.9, 0, .4],
+                    [2.8, .9, 1.4], [2.5, -.5, 2], [3.9, 0, 1.4]])
+dirs = np.array([[-.2, .9, .4], [-.5, -.5, 1], [.9, 0, .4], [-.2, .9, .4],
+                 [-.5, -.5, 1], [.9, 0, .4], [-.2, .9, .4], [-.5, -.5, 1],
+                 [.9, 0, .4]])
+colors = np.array([[1, 0, 0], [1, 0, 0], [1, 0, 0], [0, 1, 0], [0, 1, 0],
+                   [0, 1, 0], [0, 0, 1], [0, 0, 1], [0, 0, 1]])
+radius = .5
+height = 1
+
+###############################################################################
+# In order to see how cylinders are made, we set different resolutions (number
+# of sides used to define the bases of the cylinder) to see how it changes the
+# surface of the primitive.
+
+cylinders_8 = actor.cylinder(centers[:3], dirs[:3], colors[:3], radius=radius,
+                             heights=height, capped=True, resolution=8)
+cylinders_16 = actor.cylinder(centers[3: 6], dirs[3: 6], colors[3: 6],
+                              radius=radius, heights=height, capped=True,
+                              resolution=16)
+cylinders_32 = actor.cylinder(centers[6: 9], dirs[6: 9], colors[6: 9],
+                              radius=radius, heights=height, capped=True,
+                              resolution=32)
+
+###############################################################################
+# Next, we set up a new scene to add and visualize the actors created.
+
+scene = window.Scene()
+
+scene.add(cylinders_8)
+scene.add(cylinders_16)
+scene.add(cylinders_32)
+
+interactive = False
+
+if interactive:
+    window.show(scene)
+
+window.record(scene, size=(600, 600), out_path='viz_poly_cylinder.png')
+
+###############################################################################
+# Visualize the surface geometry representation for the object.
+
+cylinders_8.GetProperty().SetRepresentationToWireframe()
+cylinders_16.GetProperty().SetRepresentationToWireframe()
+cylinders_32.GetProperty().SetRepresentationToWireframe()
+
+if interactive:
+    window.show(scene)
+
+window.record(scene, size=(600, 600), out_path='viz_poly_cylinder_geom.png')
+
+###############################################################################
+# Then we clean the scene to render the boxes we will use to render our
+# SDF-based actors.
+
+scene.clear()
+
+###############################################################################
+# Cylinder using SDF
+# ================
+# We will use the ray marching algorithm to render the SDF primitive using
+# shaders. Ray marching is a technique where you step along a ray in order to
+# find intersections with solid geometry. Objects in the scene are defined by
+# SDF, and because we don’t use polygonal meshes it is possible to define
+# perfectly smooth surfaces and allows a faster rendering in comparison to
+# polygon-based modeling.
```

Comment:
is there a paper to cite here @tvcastillod @guaje ?

Reply:
Yes, here's the reference in MLA format:

Hart, John C. "Sphere tracing: A geometric method for the antialiased ray tracing of implicit surfaces." The Visual Computer 12.10 (1996): 527-545.